Hw04 auth

KovalchukAnastasia opened the pull request.

KovalchukAnastasia committed:
add_hw_04

KovalchukAnastasia committed:
add_errors

Harryk13 commented on controllers/contacts.js:
```diff
@@ -17,7 +23,8 @@ const getById = async (req, res) => {
 
 const add = async (req, res) => {
-  const newContact = await Contact.create(req.body);
+  const { _id: owner } = req.user;
```
а що на рахунок захисту роутів на редагування та видалення, отримання по ід?

KovalchukAnastasia replied:
1. захист роутів через міделвару authenticate
2. додала owner в контролери

Harryk13 replied:
Читаємо документацію та вірно використовуємо методи
https://mongoosejs.com/docs/api/model.html#model_Model-findById

KovalchukAnastasia replied:
1. findByIdAndUpdate change to findOneAndUpdate
2. findByIdAndRemove change to findOneAndDelete

KovalchukAnastasia committed:
fix_contacts_controllers

KovalchukAnastasia committed:
fix_method

KovalchukAnastasia committed:
fix_methods

1. findByIdAndUpdate change to findOneAndUpdate
2. findByIdAndRemove change to findOneAndDelete
